app.py

```diff
@@ -39,7 +47,7 @@ def static_file(path):
 # AOAI Integration Settings
 AZURE_OPENAI_RESOURCE = os.environ.get("AZURE_OPENAI_RESOURCE")
 AZURE_OPENAI_MODEL = os.environ.get("AZURE_OPENAI_MODEL")
-AZURE_OPENAI_KEY = os.environ.get("AZURE_OPENAI_KEY")
+AZURE_OPENAI_KEY = openai_token.token if USE_RBAC else os.environ.get("AZURE_OPENAI_KEY")
```

Comment:
This token will need to manually refreshed every so often if using old SDK, see bottom ofhttps://blog.pamelafox.org/2023/09/best-practices-for-openai-chat-apps-go.html

Reply:
Upgrade `openai` to V1.